### PR TITLE
ODH-1442: Running DAGs per hosts for Airflow

### PR DESCRIPTION
**Pull request by chermenin**

### CODE REVIEW

File: src/applications/basic_analytics/airflow/worker_dag_driver.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from pyspark.sql.types import *
+
+from common.kafka_pipeline import KafkaPipeline
+from common.basic_analytics.basic_analytics_processor import BasicAnalyticsProcessor
+from common.basic_analytics.aggregations import *
+from util.utils import Utils
+
+
+class AirflowWorkerDag(BasicAnalyticsProcessor):
+    def _process_pipeline(self, read_stream):
```
Comment (ltv12) [inline src/applications/basic_analytics/airflow/worker_dag_driver.py]: Please add documentation for class

File: src/common/basic_analytics/aggregations.py [hunk 2]
```diff
@@ -63,37 +63,51 @@ class Avg(Aggregation):
...
+        return grouped_dataframe.agg(count("*").alias("value"))
+
+
+class DistinctCount(Aggregation):
+    """
+    Returns an approximate distinct count of ``_aggregation_field``
+    """
+
+    def aggregate(self, grouped_dataframe):
+        return grouped_dataframe.agg(approx_count_distinct(self._aggregation_field).alias("value"))
```
Comment (ltv12) [inline src/common/basic_analytics/aggregations.py]: Please change name generation for metric name generation from 'ditstinctcount' to 'distinct_count', 
It  will add readability.

File: src/applications/basic_analytics/airflow/worker_dag_driver.py (hunk 1 above)
Comment (ltv12) [inline src/applications/basic_analytics/airflow/worker_dag_driver.py]: Please add documentation for class

File: src/common/basic_analytics/aggregations.py (hunk 2 above)
Comment (chermenin) [reply src/common/basic_analytics/aggregations.py]: Fixed for all metrics by adding the `__convert_to_underlined` method.

**Pull request merged**